<a href="https://colab.research.google.com/github/duwain/Project-2/blob/main/Project_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Project 2 Part 1**

* Author: Duwain Hofmeester

## **First Choice Dataset**

##### **Imports**

In [ ]:
# Import required packages
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
from google.colab import drive

#### **Load Dataset**

In [ ]:
drive.mount("/content/drive", force_remount=True)

Mounted at /content/drive


In [ ]:
# Save the filepath and load the data as "df_titanic"
fpath= "/content/drive/MyDrive/Project 2/First choice Dataset/Car_Insurance_Claim.csv"
df = pd.read_csv(fpath)

In [ ]:
df.head()

ID    AGE  GENDER      RACE DRIVING_EXPERIENCE    EDUCATION  \
0  569520    65+  female  majority               0-9y  high school   
1  750365  16-25    male  majority               0-9y         none   
2  199901  16-25  female  majority               0-9y  high school   
3  478866  16-25    male  majority               0-9y   university   
4  731664  26-39    male  majority             10-19y         none   

          INCOME  CREDIT_SCORE  VEHICLE_OWNERSHIP VEHICLE_YEAR  MARRIED  \
0    upper class      0.629027                1.0   after 2015      0.0   
1        poverty      0.357757                0.0  before 2015      0.0   
2  working class      0.493146                1.0  before 2015      0.0   
3  working class      0.206013                1.0  before 2015      0.0   
4  working class      0.388366                1.0  before 2015      0.0   

   CHILDREN  POSTAL_CODE  ANNUAL_MILEAGE VEHICLE_TYPE  SPEEDING_VIOLATIONS  \
0       1.0        10238         12000.0        sedan                    0   
1       0.0        10238         16000.0        sedan                    0   
2       0.0        10238         11000.0        sedan                    0   
3       1.0        32765         11000.0        sedan                    0   
4       0.0        32765         12000.0        sedan                    2   

   DUIS  PAST_ACCIDENTS  OUTCOME  
0     0               0      0.0  
1     0               0      1.0  
2     0               0      0.0  
3     0               0      0.0  
4     0               1      1.0

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 19 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   ID                   10000 non-null  int64  
 1   AGE                  10000 non-null  object 
 2   GENDER               10000 non-null  object 
 3   RACE                 10000 non-null  object 
 4   DRIVING_EXPERIENCE   10000 non-null  object 
 5   EDUCATION            10000 non-null  object 
 6   INCOME               10000 non-null  object 
 7   CREDIT_SCORE         9018 non-null   float64
 8   VEHICLE_OWNERSHIP    10000 non-null  float64
 9   VEHICLE_YEAR         10000 non-null  object 
 10  MARRIED              10000 non-null  float64
 11  CHILDREN             10000 non-null  float64
 12  POSTAL_CODE          10000 non-null  int64  
 13  ANNUAL_MILEAGE       9043 non-null   float64
 14  VEHICLE_TYPE         10000 non-null  object 
 15  SPEEDING_VIOLATIONS  10000 non-null  

In [8]:
# Display the number of rows and columns for the dataframe
df.shape
print(f'There are {df.shape[0]} rows, and {df.shape[1]} columns.')
print(f'The rows represent {df.shape[0]} observations, and the columns represent {df.shape[1]-1} features and 1 target variable.')

There are 10000 rows, and 19 columns.
The rows represent 10000 observations, and the columns represent 18 features and 1 target variable.


### **1. Source of data**
- Kaggle (https://www.kaggle.com/datasets/sagnik1511/car-insurance-data?resource=download}

### **2. Brief description of data**
- We have to determine the outcome of a customer's behaviour based on the different features.

-    ID                                int64  
-    AGE                               object
-    GENDER                            object
-    RACE                              object
-    DRIVING_EXPERIENCE                object
-    EDUCATION                         object
-    INCOME                            object
-    CREDIT_SCORE                      float64
-    VEHICLE_OWNERSHIP                 float64
-    VEHICLE_YEAR                      object
-    MARRIED                           float64
-    CHILDREN                          float64
-    POSTAL_CODE                       int64  
-    ANNUAL_MILEAGE                    float64
-    VEHICLE_TYPE                      object
-    SPEEDING_VIOLATIONS               int64  
-    DUIS                              int64  
-    PAST_ACCIDENTS                    int64  
-    OUTCOME                           float64

### **3. What is the target?**
- OUTCOME

### **4. What does one row represent? (A person? A business? An event? A product?)**
- Each row in the data provides relavant information about the customer.

### **5. Is this a classification or regression problem?**
- This is a classification problem

### **6. How many features does the data have?**
- 19 features of which 1 is the target (OUTCOME)

### **7. How many rows are in the dataset?**
- There are 10000 rows

### **8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?**

- CREDIT SCORE and ANNUAL MILEAGE feature has missing values
- unnecessary columns can be dropped such as ID

## **Second Choice Dataset**

In [9]:
# Save the filepath and load the data as "df_titanic"
fpath= "/content/drive/MyDrive/Project 2/Second choice Dataset/healthcare-dataset-stroke-data.csv"
df = pd.read_csv(fpath)

In [10]:
df.head()

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5110 entries, 0 to 5109
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 5110 non-null   int64  
 1   gender             5110 non-null   object 
 2   age                5110 non-null   float64
 3   hypertension       5110 non-null   int64  
 4   heart_disease      5110 non-null   int64  
 5   ever_married       5110 non-null   object 
 6   work_type          5110 non-null   object 
 7   Residence_type     5110 non-null   object 
 8   avg_glucose_level  5110 non-null   float64
 9   bmi                4909 non-null   float64
 10  smoking_status     5110 non-null   object 
 11  stroke             5110 non-null   int64  
dtypes: float64(3), int64(4), object(5)
memory usage: 479.2+ KB


In [13]:
# Display the number of rows and columns for the dataframe
df.shape
print(f'There are {df.shape[0]} rows, and {df.shape[1]} columns.')
print(f'The rows represent {df.shape[0]} observations, and the columns represent {df.shape[1]-1} features and 1 target variable.')

There are 5110 rows, and 12 columns.
The rows represent 5110 observations, and the columns represent 11 features and 1 target variable.


### **1. Source of data**
- Kaggle (https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset}

### **2. Brief description of data**
- This dataset is used to predict whether a patient is likely to get stroke based on the input parameters like gender, age, various diseases, and smoking status.

 -   id: unique identifier                    int64    
 -   gender: "Male", "Female" or "Other"      object
 -   age:age of the patient                   float64
 -   hypertension: 0 if the patient doesn't have hypertension, 1 if the patient has hypertension                     int64  
 -   heart_disease: 0 if the patient doesn't have any heart diseases, 1 if the patient has a heart disease           int64  
 -   ever_married: "No" or "Yes"              object
 -   work_type: "Govt_jov", "Never_worked", "Private" or "Self-employed"                              object
 -   Residence_type: "Rural" or "Urban"       object
 -   avg_glucose_level: average glucose level in blood  float64
 -   bmi:body mass index                      float64
 -   smoking_status: "formerly smoked", "never smoked", "smokes" or "Unknown"*                                    object
 -   stroke: 1 if the patient had a stroke or 0 if not                int64

### **3. What is the target?**
- STROKE

### **4. What does one row represent? (A person? A business? An event? A product?)**
- Each row in the data provides relavant information about the patient.

### **5. Is this a classification or regression problem?**
- This is a Classification problem

### **6. How many features does the data have?**
- 12 features of which 1 is the target (STROKE)

### **7. How many rows are in the dataset?**
- There are 5110 rows

### **8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?**

- Missing values need to be addressed.
